abel/direct.py

```diff
@@ -219,9 +219,19 @@ def _pyabel_direct_integral(f, r, correction, int_func=np.trapz):
         f_r = (f[:, 1:] - f[:, :N1-1])/np.diff(r)[None, :]
 
         for i, row in enumerate(f):  # loop over rows (z)
-            out[i, :-1] += I_sqrt[II+1 == JJ]*f_r[i] \
+            corr  = I_sqrt[II+1 == JJ]*f_r[i] \
```

Comment:
This fails for the case r == 0 because of r[1:]/r[:-1], so line 21 r[0] = 1.0e-16 is still required.

Reply:
This code may need to be inside the else block?

Reply:
`r[1:]/r[:-1]` does have a divide-by-zero for the first element when `r[0] = 0`. So, the first element of the `corr` array is `np.nan` when `r[0]` is `0`. But, on line 226, we check to see if `r[0]` is close to `0` and don't use the first element of `corr` in this case.

Reply:
The evaluation, although not used, creates a warning error